Doc/library/sqlite3.rst

```diff
@@ -1201,6 +1233,30 @@ Connection objects
    .. _SQLite limit category: https://www.sqlite.org/c3ref/c_limit_attached.html
 
 
+   .. method:: getconfig(op, /)
+
+      Query a boolean connection configuration option.
+
+      :param int op:
+         A :ref:`SQLITE_DBCONFIG code <sqlite3-dbconfig-constants>`.
+
+      :rtype: bool
+
+      .. versionadded:: 3.12
+
+   .. method:: setconfig(op, enable=True, /)
```

Comment:
```suggestion
   .. method:: setconfig(op, *args)
```

We could make the spec slightly more vague, in order to not lock this to boolean options only.

Reply:
I'd prefer the more specific arg spec here, I think it's useful for users to know the specific meaning of the optional arg

Reply:
Per now, I've only added support for the boolean options. There are two other options (both very low-level and IMO not worth exposing); one that takes a string as an argument, and another that takes a C buffer (a la a contiguous `memoryview`) and two integers. Also, we do of course not know the arg spec of future options.

Possible scenario: in Python 3.15, we'll add support for the option that takes a `str` as an argument. One possibility is to just document the two different variants (a la the `range` docs):


```
   .. method:: setconfig(op, enable=True, /)
               setconfig(op, name, /)
```

... or:

```
   .. method:: setconfig(op, enable=True, /)
               setconfig(sqlite3.SQLITE_DBCONFIG_MAINDBNAME, name, /)
```

Reply:
I think we should present the signature as it currently exists in the docs, and worry about changing it when we actually change it, which will probably not be for at least several versions. But I don't feel strongly here; easy either way :)